rskj-core/src/main/java/org/ethereum/core/Transaction.java

```diff
@@ -107,27 +118,107 @@ protected Transaction(byte[] rawData) {
+                }
+                this.nonce = Collections.unmodifiableList(nonce);
+            } else {
+                this.nonce = Collections.singletonList(nullToZeroArray(transaction.get(1).getRLPData()));
+            }
+            this.gasPrice = RLP.parseCoinNonNullZero(transaction.get(2).getRLPData());
+            this.gasLimit = transaction.get(3).getRLPData();
+            this.receiveAddress = RLP.parseRskAddress(transaction.get(4).getRLPData());
+            this.value = RLP.parseCoinNullZero(transaction.get(5).getRLPData());
+            this.data = transaction.get(6).getRLPData();
+            this.ownerCount = RLP.decodeInt(nullToZeroArray(transaction.get(7).getRLPData()), 0);
+
+            if (transaction.get(8) instanceof RLPList) {
+                RLPList signaturesField = (RLPList) transaction.get(8);
+                if (signaturesField.isEmpty() || signaturesField.get(0) instanceof RLPList) {
+                    List<ECDSASignature> signatures = new ArrayList<>();
+                    byte chainId = 0;
+                    boolean chainIdUnset = true;
+                    for (RLPElement e : signaturesField) {
+                        if (e instanceof RLPList) {
+                            RLPList element = (RLPList) e;
+                            if (element.size() != 3) {
+                                throw new IllegalArgumentException("Signature field must be made up of 3 elements");
+                            }
+                            byte[] vData = element.get(0).getRLPData();
+                            byte[] r = element.get(1).getRLPData();
+                            byte[] s = element.get(2).getRLPData();
+                            ECDSASignature signature = parseSignature(vData, r, s);
+                            if (signature != null) {
+                                if (chainIdUnset) {
+                                    chainIdUnset = false;
+                                    chainId = extractChainIdFromV(vData[0]);
+                                }
+                                signatures.add(signature);
+                            }
+                        } else {
+                            throw new IllegalArgumentException("Signature field must be a list");
+                        }
+                    }
+                    if (signatures.isEmpty()) {
+                        this.chainId = 0;
+                        logger.trace("RLP encoded tx is not signed!");
+                    } else {
+                        this.chainId = chainId;
+                    }
+                    this.signatures = Collections.unmodifiableList(signatures);
+                } else {
+                    if (signaturesField.size() != 3) {
+                        throw new IllegalArgumentException("Signature field must be made up of 3 elements");
+                    }
+                    byte[] vData = signaturesField.get(0).getRLPData();
```

Comment:
similar code duplicated before, better to encapsulate signature parsing in a separate method.